File: 10.my-calendar-i.py
import bisect 
class MyCalendar:
    
    def __init__(self):
        self.cal=[]
       
    
    def book(self, start: int, end: int) -> bool:
        for a in self.cal:
            if (max(a[0], start) < min(a[1], end)):
                return False
        self.cal.append([start, end])
        return True;

        

File: 10.my-calendar-i.cpp
class MyCalendar {
    private:
        vector<pair<int, int>> cal;
    public:
        MyCalendar() {}

        bool book(int start, int end) {
            for (auto a : cal) {
                if (max(a.first, start) < min(a.second, end)) return false;
            }
            cal.push_back({start, end});
            return true;
        }


};
